src/measure_theory/measure/measure_space.lean

```diff
@@ -1824,6 +1824,61 @@ begin
     (lt_of_le_of_lt (le_iff'.1 h _) (C.finite i)), C.spanning⟩⟩,
 end
 
+section disjointed
+
+variable [measurable_space α]
+
+/-- Given `S : μ.finite_spanning_sets_in {s | measurable_set s}`,
+`disjointed_finite_spanning_sets_in` provides a `finite_spanning_sets_in {s | measurable_set s}`
+such that its underlying sets are pairwise disjoint. -/
+def disjointed_finite_spanning_sets_in (μ : measure α)
+  (S : μ.finite_spanning_sets_in {s | measurable_set s}) :
+   μ.finite_spanning_sets_in {s | measurable_set s} :=
+⟨disjointed S.set, measurable_set.disjointed S.set_mem,
+  λ n, lt_of_le_of_lt (measure_mono (disjointed_subset S.set n)) (S.finite _),
+  S.spanning ▸ Union_disjointed⟩
+
+lemma disjoint_disjointed_finite_spanning_sets_in {μ : measure α}
+  (S : μ.finite_spanning_sets_in {s | measurable_set s}) :
+  pairwise (disjoint on (disjointed_finite_spanning_sets_in μ S).set) :=
+disjoint_disjointed _
+
+private lemma exists_eq_finite_spanning_sets_in
+  (μ ν : measure α) [sigma_finite μ] [sigma_finite ν] :
+  ∃ (S : μ.finite_spanning_sets_in {s | measurable_set s})
+    (T : ν.finite_spanning_sets_in {s | measurable_set s}), S.set = T.set :=
```

Comment:
I don't understand why this is an existential - it looks like you're able to provide a direct construction?

Reply:
Would it be better if I created two defs and a lemma that say the underlying sets of the two defs are equal?

Reply:
I think it might be, although I haven't yet looked at what `finite_spanning_sets_in` is

Reply:
I've changed the statements as definitions now. Unfortunately, we have to make two definitions since `nat.unpair` is not commutative